### PR TITLE
Added Post/Thread CRUD

### PR DESCRIPTION
**Pull request by staldcunha**
- all basic crud endpoints are tested and validated. 
- in createPost, I commented out the image uploading for now. This is beacause of the assumption that posts do not need a required image.

### CODE REVIEW

File: backend/controllers/post.controller.ts [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    const validatedPostData = PostCreateSchema.parse(req.body);
+
+    // if (!req.file) {
+    //   throw new AppError(
+    //     AppErrorName.FILE_UPLOAD_ERROR,
+    //     "No file uploaded.",
+    //     400,
+    //     true,
+    //   );
+    // }
```
Comment (AarshShah9) [inline backend/controllers/post.controller.ts:55]: dont leave commented out code in

File: backend/controllers/post.controller.ts [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+      //
+      // await UploadToS3(req.file!, path);
+      //
+      // // Update event with image path after successful upload
+      // return prisma.post.update({
+      //   where: { id: post.id },
+      //   data: {
+      //     image: path,
+      //   },
+      // });
```
Comment (AarshShah9) [inline backend/controllers/post.controller.ts:81]: Same thing here

File: backend/routes/post.routes.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { upload } from "../utils/S3Uploader";
+import { verifyAuthentication } from "../middleware/verifyAuth";
+
+const router = express.Router();
+
+router.get("/test", postTest);
+router.use(verifyAuthentication); // Use auth middleware for all routes below
+router.get("/", getAllPosts);
+router.post("/", upload.single("file"), createPost);
+router.patch("/:id", upload.single("file"), updatePost);
```
Comment (AarshShah9) [inline backend/routes/post.routes.ts:18]: I am little confused as to if these posts will have images or not

File: backend/controllers/post.controller.ts [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+    });
+
+    if (newPost) {
+      // Post created successfully
+      res.status(201).json({
+        message: "Post created successfully",
+        data: newPost,
+      });
+    } else {
+      // Throw an error if the event creation returned an empty result
```
Comment (jcharlpretorius) [inline backend/controllers/post.controller.ts:91]: Correct any references of "event" to "post"

File: backend/controllers/post.controller.ts [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+
+// Get all Posts
+export const getAllPosts = async (req: Request, res: Response) => {
+  try {
+    const allPosts = await prisma.post.findMany();
+    res.status(200).json({
+      message: "All posts",
+      data: allPosts,
+    });
+  } catch (error) {
```
Comment (jcharlpretorius) [inline backend/controllers/post.controller.ts:30]: pass error to next() for consistency with other controllers

File: backend/controllers/post.controller.ts [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+        await deleteFromS3(existingPost.image);
+      }
+      const uniqueFileName = generateUniqueFileName(
+        req.file!.originalname,
+        postId,
+      );
+      const path = `images/post/${uniqueFileName}`;
+      await UploadToS3(req.file!, path);
+      validatedUpdatePostData.image = path;
+    }
```
Comment (AarshShah9) [inline backend/controllers/post.controller.ts:170]: remove image stuff?

**Comment by staldcunha** [comment]
- Posts can have optional images associated with it.
- Added createVerifiedPosts for posts created by organizations. Does not have a "verifiedPostStatus". Discussed with JC regarding this, will take it up with Aarsh.
- Did not test image functionality, however code logic is similar to JC's event and org controllers.

File: backend/controllers/post.controller.ts (hunk 6 above)
Comment (staldcunha) [reply backend/controllers/post.controller.ts:236]: check recent commit

**Pull request merged**